### DISCUSSION TITLE
C++ API for tiny  receiver

### DISCUSSION THREAD

**Discussion by laverdanny** [NightscoutFoundation/xDrip#3777] [Ideas]
Hi I'm a pretty new customer of Dexcom One+ system having some MCUs skills. I was wondering if it is possible to get a C++ library in order to create an autonomus receiver (Dexcom like) to collect datas from sensors and send them to phone. It would be lightweight and easy to carry, eventually waterproof and therefore suitable for mostly all situation

**Comment by Navid200** [comment]
What would be the purpose?
xDrip can collect directly from a One+ with no need to any other device.

**Reply by laverdanny** [reply]
first of all, I haven't found a way to make xDrip to work on my Nokia/HMD XR20. About that, does xDrip need the Dexcom original App ?
As second reason, a smartphone is a lot bigger and heavier than Dexcom like receiver, but I haven't found a way to retrieve receiver's logs (data) without original Dexcom App.
As third, but I haven't tested, using a smartphone implies to keep Bluetooth and app costantly running and consuming battery charge. Dexcom receiver charge lasts almost a week, it should shutdown power consumption between two communications

**Reply by Navid200** [reply]
Your original post states that your idea is to create a device that sends readings to the phone.  Now, you are saying something else.  

If you want to use a Dexcom receiver, why don't you?  It is still not clear what you want to create.

Any way, this is how xDrip can collect directly from One+:
https://navid200.github.io/xDrip/docs/Dexcom/G7.html

**Reply by laverdanny** [reply]
As I told as second reason, Dexcom receiver only shows graphically the last 24 hours. The only way to get values logged is to use Dexcom app, but I'm not keen to share that kind of infos with them. I don't like to carry smartphone all the time, and my usual activity is not good for such a device. The purpose is to replace Dexcom receiver with a similar one able to store more than the Dexcom's 3 hours, and able to send them to a more powerful device (the smatphone) to analyse them. That way I'll be not forced to carry smartphone all the time, and avoid to constantly drain its battery too

**Reply by laverdanny** [reply]
About "https://navid200.github.io/xDrip/docs/Dexcom/G7.html", I've already tried to install it on my smartphone, no way to make it work. And QRcode didn't lead anywhere neither

**Reply by Navid200** [reply]
I see what you mean now.

xDrip can only run on a smartphone.  These are xDrip's forums.  They are meant to be used to help xDrip users.
I wish you success in getting the information you need.   Perhaps different forums may be better.

**Reply by laverdanny** [reply]
is there a way to make xDrip to work on XR20 ?
I just installed 20241117.6fcec23,Android told me the App had been built for older Android versions and doesn't include last protection and privacy rules.
I forced it to install, and when started up a siilar message tells me App is for older Androids and may not work.
it's a Nokia XR20 TA-1362 running Android 14
Have you any clue to work it out ? Must I create a new issue ?

**Reply by laverdanny** [reply]
can the One+ manage two connexions ? I mean, must I disconnect Dexcom receiver to make it work ?

**Reply by Navid200** [reply]
No, you can ignore all of those messages.  

https://navid200.github.io/xDrip/docs/FAQ/UnsafeAppBlocked.html

**Reply by Navid200** [reply]
You can use both the Dexcom receiver and xDrip to collect from the same One+.
You just have to establish connectivity with xDrip before the receiver as explained in the guide.

**Reply by laverdanny** [reply]
I did but still no way to connect to One+
May you or someone else helpme to make it work ?

**Reply by Navid200** [reply]
Post a screenshot of the Dex Status page.

**Reply by laverdanny** [reply]
Dex Status page ? do you mean xDrip status ? if not, where can I find it ?

**Reply by laverdanny** [reply]
ok got it, there was a problem with localisation permission, now scanning

**Reply by laverdanny** [reply]
Now Brain State : Scanning errors 4
Sensor status : Missing QR code
which QR code ?

**Comment by laverdanny** [comment]
Now Brain State : Scanning errors 4
Sensor status : Missing QR code
which QR code ?

**Reply by Navid200** [reply]
You need to follow the guide.  I provided a link above.
It explains everything.  It shows you a QR code.  It explains how you need to scan it.

**Reply by laverdanny** [reply]
ok, I haven't found all the settings (ie bluetooth battery management) but it works now
Actually, I tried to read QR code via camera directly, and finally found an option among xDrip menus
What the QR's 3 parameters are supposed to do ?
Thanks a lot
